### ISSUE TITLE
What is a bulletin “message”?

### ISSUE THREAD

**Issue by strogonoff** [ituob/itu-ob-data#11]
When discussing ITU OB data format, we mention OB “messages”. Can we clarify the meaning of the term? What inside an OB document counts as a message?

**Comment by strogonoff** [comment]
My current understanding is as follows:

* A message represents a concrete **change** in some aspect of communications (numbering schema, etc.)
* We want to let site consumers search for changes (messages) relevant to their area of operation
* All changes described in OB issue reside under “Amendments” section. Things such as “approval of ITU recommendations” are **not** changes (messages) our users care about
* Each change described in OB has an operation code (ADD, SUP, LIR, PAR)
* NNP changes are not formally described in OB (although OB provides freeform communication texts from ITU members)
* NNP site provides numbering schema snapshot at latest point in time, but no change feed
* We may be able to obtain NNP changes in some normalized way by collaborating with NNP team

**Comment by ronaldtse** [comment]
> A message represents a concrete change in some aspect of communications (numbering schema, etc.)

A message represents in a notice with its own section in the OB. A notice is not always a "change". For example, "call-back procedures" is a notice that is repeated in every OB, it is not a "change", but it is still a message.

> “Message”, “change” and “amendment” are used interchangeably.

As described above, they are not. We could say "notice" and "message" are interchangeable. A change or amendment can generate a message. Not all changes create a message, and not all messages are about changes.

> We want to let site consumers search for changes (messages) relevant to their area of operation

Yes, let me rephrase: we want the site consumers to "filter messages by message type".

> Each change described in OB has an operation code (ADD, SUP, LIR, PAR)

These codes ONLY pertain to the particular message type. 

> NNP changes are not formally described in OB (although OB provides freeform announcement texts from ITU members)

We should not allow freeform announcement texts moving forward, but yes, maybe we need freeform for backwards compatibility.

> NNP team does have all NNP data in XML (including those mentioned in communications, maybe more). However, NNP site provides only the latest snapshot of the data, and no change feed

Unfortunately, no. The NNP XML project is not complete. You are right there is no change feed in NNP XML.

The OB team however has all the NNP change notices.

> We may be able to obtain NNP change feed, or at worst the latest state of the data, in XML by collaborating with NNP team

There is no definition of "NNP change" yet, it seems that countries just upload their new NNP XML. For example, it doesn't seem that NNP XML allows describing the "deletion" of numbering resources.

> All together, the changes (messages) we will offer site consumers will be obtained from two sources:
> * serialized in YAML, from the new OB data format (amendments section) and

Yes.

> * serialized in XML, from NNP side

I do not know if NNP side can provide the "NNP changes" in XML. Moreover, not all countries have NNP XML. We may need to do "NNP changes" in the YAML format for now?

**Comment by strogonoff** [comment]
> The OB team however has all the NNP change notices.
> We may need to do "NNP changes" in the YAML format for now?

@ronaldtse Do we have the numbering schema changes described in E.164 section in a more formalized way than what is currently available in OB issues? If not in XML, then at least something reminding a standardized set of fields & values.

**Comment by strogonoff** [comment]
> A message represents in a notice with its own section in the OB.

@ronaldtse  Let’s take 1173 as a reference example and look at Amendments -> E.118 section.

By the definition you provided, an individual message that we represent is this whole section, right? I.e., we do **not** about singling out atomic messages, each of the form `LIR | Australia | <company name & address> | 89 41 23 | <company contact>`?

**Comment by ronaldtse** [comment]
> By the definition you provided, an individual message that we represent is this whole section, right? With ITCC changes for all countries combined? I.e., we are not planning to single out atomic messages, each of the form LIR | Australia | <company name & address> | 89 41 23 | <company contact>, in foreseeable future?

Yes, of course it would be great to normalize the data in a more granular way, but "message = section" is good enough for now.

**Comment by ronaldtse** [comment]
> @ronaldtse Do we have the numbering schema changes described in E.164 section in a more formalized way than what is currently available in OB issues? If not in XML, then at least something reminding a standardized set of fields & values.

No XML right now, but for a standardized message ("standardized form") we can refer to the forms given in E.129.
